src/components/Updater.svelte

```diff
@@ -27,17 +30,19 @@
 
                 case 'Progress':
                     downloaded += event.data.chunkLength / 1000.0;
-                    ref.setAttribute(
-                        'style',
-                        `width:${Math.round((downloaded / totalDownload) * 100)}%`
-                    );
+                    completed = Math.round((downloaded / totalDownload) * 100);
+                    message = `Downloading... (${completed}%)`;
+                    ref.setAttribute('style', `width:${completed}%`);
                     break;
 
                 case 'Finished':
-                    finished = true;
+                    message = 'Installing...';
                     break;
             }
         });
+
+        message = 'Complete';
+        finished = true;
```

Comment:
This needs to come after the entire `await update?.downloadAndInstall()` call, instead of in the `Finished` event. The `Finished` event is when the _download_ is finished, not the entire process 🙃